### ISSUE TITLE
Returning detached numpy in pytorch backend 

### ISSUE THREAD

**Issue by martinsipka** [geomstats/geomstats#1072]
https://github.com/geomstats/geomstats/blob/8fd17aee1d1ef81218b8c6fe65c9c26b9a5b394e/geomstats/_backend/pytorch/autograd.py#L28

I was wondering why the conversion to numpy does take place. Is there a compatibility reason for this?

**Comment by nguigs** [comment]
Hi!

The autograd module of the backend was created for the purpose of solving e.g [geodesic shooting](https://github.com/geomstats/geomstats/blob/5681c3695b177d73e06136201ccc2036d9a01332/geomstats/geometry/connection.py#L150-L162) problem with scipy minimize method, which only takes 1d numpy arrays as input.

An other solution would be to use pytorch/tensorflow's optimizers but we would need to also refactor them. This would be useful e.g. to take derivatives of functions defined by a minimum as [in this paper](https://arxiv.org/abs/2002.03722).

**Comment by martinsipka** [comment]
Hello, 

thank you for the reply. I see now. I am going to look at this because i suspect It might be better for performance to have something without detach/clone calls inside minimize routines. Do you have some example which heavily uses geodesics shooting that I can use for benchmarking?
